rust/src/push/evaluator.rs

```diff
@@ -154,6 +158,16 @@ impl PushRuleEvaluator {
                 continue;
             }
 
+            // For backwards-compatibility the legacy mention rules are disabled
+            // only if the event contains the 'm.mentions' property.
```

Comment:
Do you mean `if` rather than `only if` here?

`A if B` means `if B, then A`.
`A only if B` means `if A, then B`.